### ISSUE TITLE
`inject` not working on i04

### ISSUE THREAD

**Issue by DominicOram** [DiamondLightSource/blueapi#647]
In testing on i04 today we had a plan like:
```python
def thaw(smargon:Smargon = inject("smargon")):
    pass
```

This failed if we called it via `run_plan()` from GDA but passed if we did `run_plan("smargon")`. I04 is loading `dodal` from the scratch area.

## Acceptance Criteria
* `inject` works correctly

**Comment by noemifrisina** [comment]
Run into something similar, is this related to #633 or a new one?

**Comment by DominicOram** [comment]
I saw that but read it as "BlueAPI should give cleaner errors if the device doesn't exist". This was that the devices do exist but blueAPI can't find them. In your case was there a `pmac` object that blueAPI failed to pick up? Did you manage to fix the issue?

**Comment by noemifrisina** [comment]
Yes the device was there, and could be found by running `blueapi -c config.yaml controller devices`, but wasn't picked up. The issue is still there, I just worked around it by explicitly passing the devices from the command line instead of relying on inject. This is not a great solution for plans that require multiple devices though

**Comment by DiamondJoseph** [comment]
`inject` only ever really worked by co-incidence, I've been trying to get [dodal #597](https://github.com/DiamondLightSource/dodal/pull/597) over the line so that `inject("foo")` can be replaced with `ixx.foo(connect=False)`, which would allow the device to be created, and `yield from ensure_connected(foo, ...)` at the start of the plan to ensure it's connected.
